symbolic-symcache/src/new/raw.rs

```diff
@@ -53,6 +53,8 @@ pub struct Header {
 pub struct Function {
     /// The functions name (reference to a [`String`]).
     pub name_idx: u32,
+    /// The compilation directory (reference to a [`String`]).
+    pub comp_dir_idx: u32,
```

Comment:
Q: so this requirements comes from the existing SymCache API, right? Would it make sense to actually save the whole filename (consisting of comp_dir + relative paths to the file the function is defined in)?

Reply:
It's a combination of the existing symcache API and the information we get from e.g. the `DebugSession` trait. A `symbolic_debuginfo::Function` has a field for the compilation directory, but not the file name. My guess is that the latter isn't always well-defined, but the former is.